### ISSUE TITLE
Dropped Queries and Processing Errors since v.2.4.2

### ISSUE THREAD

**Issue by 0x6d69636b** [grafana/loki#5942]
I use a Grafana dashboard for iptables reports from loki with queries such as:

```
count_over_time({host="$host"}|~"ip6{0,1}tables" |="IN=ens19" |="DROPPED:"[1m])
count_over_time({host="$host"}|~"iptables" |="IN=ens19" |="DROPPED:"[1m])
```
In Loki v.2.4.1 this worked without problems. Since Loki v.2.4.2 and v2.5.0 there are errors and the results cannot be displayed. This happens for a time range greater than the last 30 minutes. In v.2.4.1 longer time ranges worked without problems.

The following errors are logged:
```
level=info ts=2022-04-18T06:55:46.802680464Z caller=metrics.go:122 component=querier org_id=fake latency=fast query="count_over_time({host=\"$host\"} |~ \"iptables\" |= \"IN=ens19\" |= \"DROPPED:\"[1m])" query_type=metric range_type=range length=25m50s step=5s duration=10.499813ms status=200 limit=454 returned_lines=0 throughput=0B total_bytes=0B queue_time=106.799107ms subqueries=1
level=error ts=2022-04-18T06:58:40.969772032Z caller=retry.go:73 org_id=fake msg="error processing request" try=0 err="context canceled"
```
This is my loki config (/etc/loki/loki.yaml)
```
auth_enabled: false

server:
  http_listen_port: 3100

ingester:
  lifecycler:
    address: 127.0.0.1
    ring:
      kvstore:
        store: inmemory
      replication_factor: 1
    final_sleep: 0s
  chunk_idle_period: 5m
  chunk_retain_period: 30s
  wal:
    enabled: true
    dir: /var/lib/loki/wal

schema_config:
  configs:
  - from: 2020-05-15
    store: boltdb
    object_store: filesystem
    schema: v11
    index:
      prefix: index_
      period: 168h

storage_config:
  boltdb:
    directory: /var/lib/loki/index

  filesystem:
    directory: /var/lib/loki/chunks

limits_config:
  enforce_metric_name: false
  reject_old_samples: true
  reject_old_samples_max_age: 168h
```

**Comment by slim-bean** [comment]
The messages you pasted actually look normal to me and the first line indicates the query actually completed successfully. 

Did you also update Grafana? Are there other logs from Loki or Grafana?  What errors are you seeing and where (aside from those logs)?

**Comment by 0x6d69636b** [comment]
I'm runnig Grafana v8.4.6 (c53173ff6b). I see also errors in Grafana (no data)

![grafana_dashboard_no_data](https://user-images.githubusercontent.com/13913706/164066834-fad23a36-c67c-4e0e-8d5e-9dcf680da671.png)

```
too many outstanding requests
[Object]()
[status:429]()
[statusText:"Too Many Requests"]()
[data:Object]()
[message:"too many outstanding requests"]()
[error:"Too Many Requests"]()
[response:"too many outstanding requests"]()
[config:Object]()
[headers:Object]()
[url:"api/datasources/proxy/2/loki/api/v1/query_range?direction=BACKWARD&limit=1378&query=count_over_time(%7Bhost%3D%22aneris%22%7D%7C~%22ip6tables%22%20%7C%3D%22IN%3Dens19%22%20%7C%3D%22DROPPED%3A%22%5B1m%5D)&start=1650369353534000000&end=1650390953535000000&step=15"]()
[retry:0]()
[hideFromInspector:false]()
[message:"too many outstanding requests"]()
```
What I don't understand is that in v.2.4.1 there are no problems and the queries can be processed. Only with 2.4.2 or 2.5.0 does this behaviour occur.

**Comment by liguozhong** [comment]
"too many outstanding requests“
This error can be resolved by increasing this conf.

```yaml
query_scheduler:
  max_outstanding_requests_per_tenant: 1000
frontend:
  max_outstanding_per_tenant: 1000
```

**Comment by 0x6d69636b** [comment]
I updated my config and the message "too many outstanding requests“ is gone, however the issue not. Now I got a lot of those messages: "level=error ts=2022-06-10T17:02:33.17980133Z caller=retry.go:73 org_id=fake msg="error processing request" try=0 err="context canceled"". However, I guess my issue is a duplicate of https://github.com/grafana/loki/issues/5123

**Comment by slim-bean** [comment]
@0x6d69636b that error message is quite misleading I'm afraid. It's likely not an error at all.

What changed between the releases you mention is Loki automatically doing more query parallelization. This change is intended to make Loki faster but the side effects are what you see with too many outstanding requests

Those "context cancelled" messages can occur because Loki will run multiple subqueries in parallel, but if enough data is found to return a result and their are still subqueries running they are cancelled which leads to that message. So it's not really a problem just bad error UX

Still some work to do in Loki to make this more pleasant for people running it! Thanks for reporting issues though, it helps us make this better!
